tests/EventManagerTest.php

```diff
@@ -23,21 +23,42 @@ public function testAttachFirstListener()
         $this->assertCount(1, $listeners);
     }
 
-    public function testCallEvent()
+    /**
+     * @dataProvider triggers
+     */
+    public function testCallEvent($regexp, $attach, $called)
     {
         $eventManager = new EventManager();
-        $eventManager->attach("post-save", function ($assert) {
-            $this->assertSame("override", $assert);
```

Comment:
**PAY ATTENTION ON THIS**

in that case if a library bug miss the action call the error is never triggered

use something like:

```php
$override = false;
$eventManager->attach("post-save", function ($assert) use (&$override) {
    $override = true;
});
$eventManager->trigger("/post-save/", ["override"]); // sequential
$this->assertSame(true, $override);  // The assert is always called 
```